### PR TITLE
Remove all uses of Python apart from Rally

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Remove all uses of Python apart from Rally, to simplify and speed up our build, and reduce the amount of custom code we are maintaining. Because only Rally requires Python now, we remove `script/requirements.txt`, and the `autopep8` and `check-autopep8` Makefile targets. CI was relying on docker-compose being upgraded by the version specified in `script/requirements.txt`; this was a CI-only thing, so I've moved this to the relevant CI script.

`script/copy_package.py` is no longer needed, as snapshot packages are automatically published to storage. Non-snapshot packages do not need any modifications, and can be copied verbatim.

`script/copy-docs.py` is replaced with a simple `rsync` command. `make update-beats-docs` now copies from the beats module specified in go.mod.

`script/generate_notice.py` is replaced with a simpler "go list" command that is directly piped to `go-licence-detector`. I have copied the beats NOTICE.txt/dependencies.csv templates, rules, and overrides, with our modifications made directly in the templates. The order of dependencies in NOTICE.txt has changed, hence the file has been regenerated.

`script/latest_snapshot_version.py` is replaced with a simple shell script using `curl` and `jq`.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

N/A

## Related issues

None

### CODE REVIEW

File: script/latest_snapshot_version.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+#
+# Find the latest snapshot version for the specified branch.
+#
+# Example usage: ./latest_snapshot_version.py 7.x
+
+BRANCH=$1
```
Comment (kruskall) [inline script/latest_snapshot_version.sh:7]: **suggestion**: should we guard against `BRANCH` being empty and return an error ?

File: script/latest_snapshot_version.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# The snapshots API has not yet been updated to use "main".
+if [ "$BRANCH" = "main" ]; then
+  BRANCH=master
+fi
+
+curl --silent "https://snapshots.elastic.co/latest/$BRANCH.json" | jq -r .version
```
Comment (kruskall) [inline script/latest_snapshot_version.sh:14]: **issue**: curl is suppressing errors. I think we should only ignore stdout and let stderr print the errors (`jq` will fail anyway but at least we would have the error message)

Something like `curl --silent --show-error` should be okay

Comment (kruskall) [inline script/latest_snapshot_version.sh:14]: **issue**: `jq` doesn't return a proper exit code. We should use `-e` to set the exit code and try to handle empty input or missing key.

To reproduce:

```
~ % echo "" | jq -r .version
~ % echo $?
0
```

Fix attempt using `-e -n`:

```
~ % echo "" | jq -e -n -r 'input.version'
jq: error (at <stdin>:1): break
~ % echo $?
1
```

**Review by kruskall — CHANGES_REQUESTED** [review]
Great change! Just a minor issue in the shell script

**Comment by axw** [comment]
Thanks @kruskall, good catches! Updated with your suggestions.

**Review by kruskall — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**